headers/types/types.h

```diff
@@ -2,6 +2,7 @@
 #define HEADERS_TYPES_H_
 
 #include "common/common.h"
+#include "common/sound.h"
```

Comment:
I'm not a fan of this import structure. Can we `#include sound.h` from `common.h`? I think circular imports can be avoided by moving `struct thread` and `struct thread_info` to `util.h` and then including `enums.h`, `file_io.h`, and `util.h` individually in `sound.h` rather than the full `common.h`.

Reply:
Yeah, this works.